GEOS-CF version 2

CarlMalings opened the pull request:
Added files to describe the new version (version 2, or v2) of the GEOS-CF dataset.

Also, modified the files describing the GEOS-CF version 1 dataset to replace a dead link with a live one.

CarlMalings committed:
Create NASA_GEOS-CF_v2.libsonnet

Added new description file for all GEOS-CF v2 collections.

CarlMalings committed:
Update NASA_GEOS-CF_v1_fcst_htf.jsonnet

Update GEOS-CF website link

CarlMalings committed:
Update NASA_GEOS-CF_v1_fcst_tavg1hr.jsonnet

Update GEOS-CF Website link

CarlMalings committed:
Update NASA_GEOS-CF_v1_rpl_htf.jsonnet

Update GEOS-CF Website link

CarlMalings committed:
Update NASA_GEOS-CF_v1_rpl_tavg1hr.jsonnet

Update GEOS-CF Website link

CarlMalings committed:
Add catalog files for GEOS-CF v2

Added files to the catalog to describe new collections for version 2 of the GEOS-CF system. These files were duplicated and modified from corresponding files for the version 1 collection. Files added are:
NASA_GEOS-CF_v2.libsonnet - Overall description of the dataset
NASA_GEOS-CF_v2_tavg1hr.libsonnet - Descriptions of bands in the one-hour-average collections
NASA_GEOS-CF_v2_ana_tavg1hr.jsonnet - Description of the one-hour-average analysis collection
NASA_GEOS-CF_v2_fcst_tavg1hr.jsonnet - Description of the one-hour-average forecast collection
NASA_GEOS-CF_v2_ana_htf.jsonnet - Description of the 15-minute-instantaneous analysis collection
NASA_GEOS-CF_v2_fcst_htf.jsonnet - Description of the 15-minute-instantaneous forecast collection

CarlMalings committed:
Update catalog.jsonnet

Added NASA GEOS-CF v2 datasets (4 total)

CarlMalings committed:
Upload Examples Files for GEOS-CF v2

Uploading example files fro GEOS-CF v2 image collections:
NASA_GEOS-CF_v2_ana_tavg1hr.js
NASA_GEOS-CF_v2_ana_htf.js
NASA_GEOS-CF_v2_fcst_tavg1hr_preview.js
NASA_GEOS-CF_v2_fcst_tavg1hr.js
NASA_GEOS-CF_v2_fcst_htf.js

CarlMalings committed:
Add Preview Files for GEOS-CF v2

Added new previous files for GEOS-CF v2 image collections

CarlMalings committed:
Merge branch 'google:main' into main

simonff commented on catalog/NASA/NASA_GEOS-CF_v2.libsonnet:
```diff
@@ -0,0 +1,53 @@
+{
+  description: |||
+    The Goddard Earth Observing System Composition Forecast (GEOS-CF) system is
+    a high-resolution (0.25&deg;) global constituent prediction system from
```
"high-resolution" is an ambiguous term - this is indeed high for forecasts, but low for satellite imagery. Is there a more targeted term we could use?

CarlMalings replied:
I removed the  "high resolution", instead just stating the resolution in the description.

simonff commented on catalog/NASA/NASA_GEOS-CF_v2.libsonnet:
```diff
@@ -0,0 +1,53 @@
+    [GEOS IT](https://gmao.gsfc.nasa.gov/gmao-products/geos-it/), direct assimilation 
+    of stratospheric profiles and column ozone from NASA Aura OMI and MLS instruments, 
+    and updates to the anthropogenic emission source. Evaluation of these updates is 
+    ongoing; results will be added into this description as they become available.
```
Do you have any known limitations that we can mention here? AI Agents are increasingly using these dataset descriptions to reason about dataset suitability, and the more common-sense wisdom we can mention in this text, the better.

CarlMalings committed:
Revise GEOS-CF v2 description

Updated the description of GEOS-CF v2 to provide more specificity about the horizontal resolution and to discuss some preliminary performance results.

dpencosk commented:
A few comments from the engineer working on ingesting the data:

**ana/tavg1hr**
The first asset for a day starts at 0030z (e.g., YYYYMMDD_0030z ). The time increments by exactly 1 hour for each subsequent asset (e.g., _0130z , _0230z , _0330z , up to _2330z ).
[Example](https://code.earthengine.google.com/10af82506124a40616dc1b1b48831505)

**fcst/htf** 
Each daily model run is initialized at 09:00z . The first forecast asset starts exactly 15 minutes later at 0915z (e.g., 20260101_09z-20260101_0915z ). The forecast target time increments by exactly 15 minutes for each subsequent asset (e.g., _0915z , _0930z , _0945z , _1000z ).
[Example](https://code.earthengine.google.com/fca219c48a4ef9a5664d51f6c578285d)

**fcst/tavg1hr** 
Each daily model run is initialized at 09:00z . Because tavg1hr represents a 1-hour time average, the first asset averages the window from 09:00z to 10:00z , centered exactly at
0930z (e.g., 20260101_09z-20260101_0930z ). The forecast target time increments by exactly 1 hour for each subsequent asset (e.g., _0930z , _1030z , _1130z , _1230z ).
[Example](https://code.earthengine.google.com/67f388445d758c79aad3641c16cbf29f)

CarlMalings committed:
Merge branch 'google:main' into main

CarlMalings committed:
Update NASA_GEOS-CF ana/tavg1hr example script

Updated NO2 image timestamp to correspond to half-hour increment.

CarlMalings committed:
Update NASA_GEOS-CF fcst/htf example script

Changed the timestamp to correspond to the correct initialization time.

CarlMalings committed:
Update NASA_GEOS-CF fcst/htf example script

Corrected timestamp to correspond to valid initialization time.

CarlMalings committed:
Add 'gee:status' field to NASA_GEOS-CF_v2_ana_htf.jsonnet

Set 'gee:status' to 'beta' as recommended by automated checker.

CarlMalings committed:
Add 'gee:status' field to NASA GEOS-CF_v2_ana_tavg.jsonnet

Set 'gee:status' to 'beta' as recommended by automated checker.

CarlMalings committed:
Update NASA_GEOS-CF_v2_fcst_htf.jsonnet

Set 'gee:status' to 'beta' as recommended by automated checker.

CarlMalings committed:
Add 'gee:status' field to NASA_GEOS-CF_v2_fcast_tavg1hr.jsonnet

Set 'gee:status' to 'beta' as recommended by automated checker.

CarlMalings commented:
I updated the examples as recommended, and added 'gee:status':'beta' to the collections as recommended by the automatic checker.

dpencosk commented on catalog/NASA/NASA_GEOS-CF_v2_ana_htf.jsonnet:
```diff
@@ -0,0 +1,123 @@
+      {
+        name: 'PM25_RH35',
+        description: 'Particulate matter with diameter below 2.5 um mass at RH 35% (includes water) mass concentration in dry air',
+        'gee:units': 'ug m-3',
```
Let's add "density_ug_per_m_cubed: 'ug/m^3'" to catalog/units.libsonnet, next to the other density units.

CarlMalings replied:
I'm worried I didn't do this correctly, I am getting errors from the checker that the new units I defined are not being recognized.

dpencosk replied:
Ah, we also need to add them to checker/units.py. We have an outstanding TODO to read the units from libsonnet file instead of duplicating the list :)

CarlMalings replied:
OK, I think I fixed that.

dpencosk commented on catalog/NASA/NASA_GEOS-CF_v2_fcst_htf.jsonnet:
```diff
@@ -0,0 +1,136 @@
+      {
+        name: 'PM25_RH35',
+        description: 'Particulate matter with diameter below 2.5 um mass at RH 35% (includes water) mass concentration in dry air',
+        'gee:units': 'ug m-3',
```
Use the new density constant here, too.

dpencosk commented on catalog/NASA/NASA_GEOS-CF_v2_tavg1hr.libsonnet:
```diff
@@ -0,0 +1,474 @@
+      {
+        name: 'PM10_RH35',
+        description: 'Particulate matter with diameter less than 10 um mass at RH 35% (includes water) mass concentration in dry air',
+        'gee:units': 'ug m-3',
```
And the new density constant can also go on all the PM*_RH* bands.

dpencosk commented on catalog/NASA/NASA_GEOS-CF_v2_tavg1hr.libsonnet:
```diff
@@ -0,0 +1,474 @@
+      {
+        name: 'PHIS',
+        description: 'Surface geopotential height',
+        'gee:units': 'm^2 s-2',
```
Can use units.squared_meter_per_squared_second

dpencosk commented on catalog/NASA/NASA_GEOS-CF_v2_tavg1hr.libsonnet:
```diff
@@ -0,0 +1,474 @@
+      {
+        name: 'TotCol_BrO',
+        description: 'Total column density of bromine monoxide (MW=95.9 g mol-1)',
+        'gee:units': '1.0e15 molec cm-2',
```
Maybe this could also be a constant, e.g. "column_density_for_trace_gases"? If not, make these "/ cm^2" for consistency with other units.

dpencosk commented on examples/NASA/NASA_GEOS-CF_v2_ana_htf_preview.js:
```diff
@@ -0,0 +1,39 @@
+// Width and height of the thumbnail image.
+var pixels = 256;
+
+var image = temp.visualize(tempVis);
```
"temp and "tempViz" are undefined. Did you mean "NO2" for "temp"?

dpencosk commented on examples/NASA/NASA_GEOS-CF_v2_ana_tavg1hr_preview.js:
```diff
@@ -0,0 +1,39 @@
+// Width and height of the thumbnail image.
+var pixels = 256;
+
+var image = temp.visualize(tempVis);
```
Same temp/NO2 issue here.

dpencosk commented on examples/NASA/NASA_GEOS-CF_v2_fcst_htf_preview.js:
```diff
@@ -0,0 +1,39 @@
+// Width and height of the thumbnail image.
+var pixels = 256;
+
+var image = temp.visualize(tempVis);
```
Same NO2/temp issue here.

dpencosk commented on examples/NASA/NASA_GEOS-CF_v2_fcst_tavg1hr_preview.js:
```diff
@@ -0,0 +1,39 @@
+// Width and height of the thumbnail image.
+var pixels = 256;
+
+var image = temp.visualize(tempVis);
```
Same NO2/temp issue here.

dpencosk commented on catalog/NASA/NASA_GEOS-CF_v2_fcst_htf.jsonnet:
```diff
@@ -0,0 +1,136 @@
+  id: id,
+  'gee:type': ee_const.gee_type.image_collection,
+  'gee:status':'beta',
+  title: 'GEOS-CF fcst htf v2: Goddard Earth Observing System Composition Forecast',
```
How are the existing v1s related to v2? Is there a 1:1 link between each of the four v1 datasets to one of these v2s? If so, we should mark the v1s as deprecated and use version links to point them forward (see examples in catalog/TOMS, specifically how TOMS_MERGED_versions.libsonnet is used in both the v3 and v4 jsonnet files)

CarlMalings replied:
There are a few differences, there are different variables in the version 1 and version 2 collections, and version 2 will not be back-filled to cover the time period of version 1. Basically, we don't see version 2 as a strict replacement for version 1 in the period they overlap, and version 1 will cover further back in time. Let me know how you think would be best to handle this.

dpencosk replied:
I see, then maybe just include a quick summary in the common descriptions how the v1s and v2s are related.

CarlMalings replied:
There was already a description of the changes in the version 2 description. I also added a note to version 1.

CarlMalings committed:
Correcting error temp and tempVis

Replacing temp with NO2 and tempVis with NO2vis

CarlMalings committed:
Correcting error temp and tempVis

Replacing temp with NO2 and tempVis with NO2vis

CarlMalings committed:
Correcting error temp and tempVis

Replacing temp with NO2 and tempVis with NO2vis

CarlMalings committed:
Correcting error temp and tempVis

Replacing temp with NO2 and tempVis with NO2vis

CarlMalings committed:
Add microgram per cubic meter density unit

Adding new mass density unit, micrograms per meter cubed, as density_ug_per_m_cubed: 'ug/m^3'

CarlMalings committed:
Change PM25_RH35 units to use defined unit

Updated units for PM25_RH35 to use a defined unit.

CarlMalings committed:
Update PM25_RH35 units to use a defined unit.

CarlMalings committed:
Add new units

CarlMalings committed:
Use standard unit definitions

Updated units for various aerosol and gas properties to use standardized unit representations.

CarlMalings committed:
Update density unit from 'ug/m^3' to 'μg/m^3'

Use mu symbol

CarlMalings committed:
Remove density_ug_per_m_cubed unit

Removed microgram per cubic meter unit from the catalog. Redundant with already existing unit mg_per_sq_m

CarlMalings committed:
Change units for PM25_RH35

Changed to use already defined unit mg_per_sq_m

CarlMalings committed:
Change units for PM25_RH35

Changed to use already defined unit mg_per_sq_m

CarlMalings committed:
Change units for particulate matter measurements

Updated units for various particulate matter metrics from new unit density_ug_per_m_cubed to pre-defined unit mg_per_sq_m.

CarlMalings committed:
Add new unit '1.0e15 molecules/cm^2' to checker UNITS set

CarlMalings committed:
Update NASA_GEOS-CF_v1.libsonnet with v2.0 details

Added details about GEOS-CF v2.0 release and its updates.

CarlMalings committed:
Fix formatting of unit '1.0e15 molecules/cm^2'

Restored a missing comma

CarlMalings committed:
Update note for GEOS-CF v2.0 release

Trying to add bold text.

CarlMalings committed:
Update NASA_GEOS-CF_v2.libsonnet with new details

Added links to the GEOS-CF Product Page and a GMAO webpage highlighting preliminary results. Clarified improvements in atmospheric composition representation in GEOS-CF v2 compared to v1.

CarlMalings committed:
Revise GEOS-CF v1 description

Updated descriptions for GEOS-CF v1, including additional details and links to outside information, include a link to preliminary results comparing GEOS-CF v1 and v2.

CarlMalings commented:
Sorry, I thought of some additional revisions to put in the descriptions; we just put up a [website ](https://gmao.gsfc.nasa.gov/science-snapshots/improved-atmospheric-composition-forecasting-in-geos-cf-version-2/) comparing GEOS-CF v1 and v2, I wanted to add links to that.
